src/activity_item_handling.cpp

```diff
@@ -1156,12 +1157,597 @@ void activity_on_turn_blueprint_move( player_activity &, player &p )
         p.assign_activity( act_multiple_construction );
         return;
     }
+}
+
+static bool is_dig_tool_nearby( player &p )
+{
+    const tripoint abspos = g->m.getabs( p.pos() );
+    const zone_manager &mgr = zone_manager::get_manager();
+    const std::unordered_set<tripoint> &src_set = mgr.get_near(
+                zone_type_id( "LOOT_TOOLS" ), abspos );
+    const zone_type_id no_pickup( "NO_NPC_PICKUP" );
+    bool found_required = false;
+    const std::vector<tripoint> &src_sorted = get_sorted_tiles_by_distance( abspos, src_set );
+    for( const tripoint &pabs : src_sorted ) {
+        tripoint p_loc = g->m.getlocal( pabs );
+        // dont pickup from this spot, even if its got what we need.
+        if( g->check_zone( no_pickup, p_loc ) ) {
+            continue;
+        }
+        // otherwise find the right tool
+        const auto items_at = g->m.i_at( p_loc );
+        for( const item &elem : items_at ) {
+            if( elem.has_quality( quality_id( "DIG" ), 1 ) ) {
+                found_required = true;
```

Comment:
```suggestion
                return true;
```
And change the last line of this function to `return false;`.